### PR TITLE
feat: new creation design + step 1

### LINKED ISSUE #961: [Create Safe] Implement redesign Network + Name step
## What is the feature about

Implement the redesign for the Network + Name step

## The list of requirements

- Has fields for changing the network and adding a name for the safe
- The name field has an icon that shows a tooltip on hover
- Shows the progress bar

## Designs/sketches
Screenshot 2022-10-24 at 11 55 30

## Links

Figma: https://www.figma.com/file/ptTs6lDBeUuLNySroJ5PiF/Web-Master-File?node-id=466%3A10273

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #961

## How this PR fixes it

This implements the new creation/load flow layout, as well as the first step of creation. The progress bar is a placeholder as it will rely on all steps/progress and the stepper hook.

Note: the info widget is yet to be included as [it has not been QA'd](https://github.com/safe-global/web-core/pull/960).

## How to test it

Open `/demo` and observe the new design.

## Screenshots

image

image

### CODE REVIEW

**Review by katspaugh — COMMENTED** [review]
Creating custom components just to add a couple of lines of CSS to MUI seems...excessive?

I would just make one Card module, and either pass props for the header and content or export them as named exports.

File: src/components/open/CreateSafe/styles.module.css [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.back {
+  margin: var(--space-4) 0;
+}
+
+.title {
+  padding-bottom: var(--space-2);
+}
```
Comment (katspaugh) [inline src/components/open/CreateSafe/styles.module.css:7]: These one-liners could be `my={4}` and `pb={2}` respectively.

**Review by katspaugh — COMMENTED** [review]
Since we're building a new file structure anyway, I suggest we don't use the old "open" folder and put stuff under `src/components/new-safe` (to be used by both Create and Add Safe).

File: src/components/open/CreateSafe/styles.module.css (hunk 1 above)
Comment (iamacook) [reply src/components/open/CreateSafe/styles.module.css]: `Button` does not accept shorthands so I've added an `sx` to it.

File: src/components/new-safe/steps/Step1/styles.module.css [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  padding-left: var(--space-1);
+  border-left: 1px solid var(--color-border-main);
+}
+
+.chain {
+  min-width: 90px;
+  min-height: var(--space-2);
+  border-radius: 4px;
+  font-size: 12px;
+  font-weight: 700px;
```
Comment (usame-algan) [inline src/components/new-safe/steps/Step1/styles.module.css:23]:
```suggestion
  font-weight: 700;
```

File: src/components/new-safe/steps/Step1/index.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+              className: css.select,
+            }}
+            defaultValue={chainId}
+          >
+            {configs.map((chain) => (
+              <MenuItem key={chain.chainId} value={chain.chainId}>
+                <ChainIndicator chainId={chain.chainId} className={css.chain} />
+              </MenuItem>
+            ))}
+          </Select>
```
Comment (usame-algan) [inline src/components/new-safe/steps/Step1/index.tsx:80]: The Menu items look different than the network selector in the header. Could we reuse `NetworkSelector` here? The main difference seems to be the label "Network" in front.

Screenshot 2022-10-25 at 11 42 50

File: src/components/new-safe/CreateSafe/index.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  const onBack = () => {
+    router.back()
+
+    // Logic to be handled by stepper hook
+  }
+
+  // TODO: Improve layout when other widget/responsive design is ready
+  return (
+    <Grid container spacing={3}>
+      <Grid item xs={1} />
```
Comment (usame-algan) [inline src/components/new-safe/CreateSafe/index.tsx:30]: I would suggest not adding these offsets here. From the designs it looks like the form will be centered and the sidebar will be hidden.

**Review by schmanu — APPROVED** [review]
Nice!

**Pull request merged**